jni/jni/javautilities.cpp

```diff
@@ -26,29 +26,75 @@
 #include <utilities/samplemanager.h>
 #include <utilities/tablepool.h>
 #include <utilities/wavereader.h>
+#include <sys/types.h>
+#include <android/asset_manager.h>
+#include <android/asset_manager_jni.h>
 
 /* SampleManager hooks */
 
 bool JavaUtilities::createSampleFromFile( jstring aKey, jstring aWAVFilePath )
 {
-    // convert jstring describing aWAVFilePath to std::string
-    const char* s = JavaBridge::getEnvironment()->GetStringUTFChars( aWAVFilePath, NULL );
-    std::string thePath = s;
-    JavaBridge::getEnvironment()->ReleaseStringUTFChars( aWAVFilePath, s );
-
+    std::string thePath = JavaBridge::getString( aWAVFilePath );
     AudioBuffer* sampleBuffer = WaveReader::fileToBuffer( thePath );
 
     // error during loading of WAV file ?
 
     if ( sampleBuffer == NULL )
         return false;
 
-    // convert jstring describing aKey to std::string
-    s = JavaBridge::getEnvironment()->GetStringUTFChars( aKey, NULL );
-    std::string theKey = s;
-    JavaBridge::getEnvironment()->ReleaseStringUTFChars( aKey, s );
+    SampleManager::setSample( JavaBridge::getString( aKey ), sampleBuffer );
+
+    return true;
+}
+
+bool JavaUtilities::createSampleFromAsset( jstring aKey, jobject assetManager, jstring assetName )
+{
+    std::string filename = JavaBridge::getString( assetName );
```

Comment:
All new: register a .WAV file present within the .APK's assets folder directly into the SampleManager.